hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/datanode/TestDataNodeMetrics.java

```diff
@@ -190,18 +190,28 @@ public Object answer(InvocationOnMock invocationOnMock)
       DataNodeFaultInjector.set(injector);
       Path testFile = new Path("/testFlushNanosMetric.txt");
       FSDataOutputStream fout = fs.create(testFile);
+      DFSOutputStream dout = (DFSOutputStream) fout.getWrappedStream();
       fout.write(new byte[1]);
       fout.hsync();
+      DatanodeInfo[] pipeline = dout.getPipeline();
       fout.close();
+      dout.close();
+      DatanodeInfo headDatanodeInfo = pipeline[0];
+      DataNode headNode = null;
       List<DataNode> datanodes = cluster.getDataNodes();
-      DataNode datanode = datanodes.get(0);
-      MetricsRecordBuilder dnMetrics = getMetrics(datanode.getMetrics().name());
+      for (DataNode datanode : datanodes) {
```

Comment:
Could we change this to lambda like this
`      DataNode headNode = datanodes.stream()
          .filter(d -> d.getDatanodeId().equals(headDatanodeInfo))
          .findFirst().orElseGet(() -> null);
`